src/dft/descriptor.cxx

```diff
@@ -26,39 +27,74 @@ namespace detail {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::set_value(config_param param, ...) {
-    int err = 0;
     va_list vl;
     va_start(vl, param);
     switch (param) {
-        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::FORWARD_DOMAIN:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::DIMENSION:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::LENGTHS: {
+            if (values_.rank == 1) {
+                values_.dimensions = std::vector<std::int64_t>{ va_arg(vl, std::int64_t) };
+            }
+            else {
+                auto ptr = va_arg(vl, std::int64_t*);
+                std::copy(ptr, ptr + values_.rank + 1, values_.dimensions.begin());
+            }
+            break;
+        }
+        case config_param::PRECISION:
+            throw mkl::invalid_argument("DFT", "set_value", "Read-only parameter.");
+            break;
+        case config_param::INPUT_STRIDES:
         case config_param::OUTPUT_STRIDES: {
-            int64_t *strides = va_arg(vl, int64_t *);
-            if (strides == nullptr)
-                break;
-            if (param == config_param::INPUT_STRIDES)
-                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.input_strides));
-            if (param == config_param::OUTPUT_STRIDES)
-                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.output_strides));
+            auto strides = va_arg(vl, std::int64_t*);
+            if (strides == nullptr) {
+                throw mkl::invalid_argument("DFT", "set_value", "Invalid config_param argument.");
+            } else if (param == config_param::INPUT_STRIDES) {
+                std::copy(strides, strides + values_.rank + 1, values_.input_strides.begin());
+            } else if (param == config_param::OUTPUT_STRIDES){ 
+                std::copy(strides, strides + values_.rank + 1, values_.output_strides.begin());
+            }
             break;
         }
         case config_param::FORWARD_SCALE: values_.fwd_scale = va_arg(vl, double); break;
         case config_param::BACKWARD_SCALE: values_.bwd_scale = va_arg(vl, double); break;
```

Comment:
@fabiomestre Type should match precision of descriptor for forwards and backwards scale.

Reply:
Done in https://github.com/hjabird/oneMKL/pull/3/commits/ffd31ad661876cada616d1bd4d90e43f237573cf.